precompile/contract.go

```diff
@@ -109,12 +109,15 @@ func newStatefulPrecompileWithFunctionSelectors(fallback RunStatefulPrecompileFu
 	for _, function := range functions {
 		_, exists := contract.functions[string(function.selector)]
 		if exists {
-			panic(fmt.Errorf("cannot create stateful precompile with duplicated function selector: %q", function.selector))
+			return nil, fmt.Errorf("cannot create stateful precompile with duplicated function selector: %q", function.selector)
+		}
+		if function == nil {
```

Comment:
why do we add the nil check for the function? This should never happen if implemented properly and will panic later in the creation of the contract if an error is returned here, so I don't think this check is necessary.

Reply:
I thought this could be go well with the existing function check. removed it.